### PR TITLE
Check old arguments which is not supported in v2 to show error message.

### PR DESCRIPTION
**Pull request by unnonouno**
I applied #2556 to all functions which had `train` or `test` arguments.

### CODE REVIEW

File: chainer/functions/connection/n_step_lstm.py [hunk 1]
```diff
@@ -402,6 +405,14 @@ def n_step_lstm(
     Note that all input variables except first layer may have different shape
     from the first layer.
 
+    .. warning::
+
+       ``train`` and ``use_cudnn`` arguments are not supported anymore since
+       v2.
+       Instead, use ``chainer.using_config('train', train)`` and
+       ``chainer.using_config('use_cudnn', use_cudnn)``.
```
Comment (delta2323) [inline chainer/functions/connection/n_step_lstm.py]: add ", respectively".

File: chainer/functions/connection/n_step_lstm.py [hunk 2]
```diff
@@ -454,6 +465,13 @@ def n_step_lstm(
 
     """
 
+    argument.check_unexpected_kwargs(
+        kwargs, train='train argument is not supported anymore. '
+        'Use chainer.using_config',
+        use_cudnn='use_cudnn argument is not supported anymore. '
+        'Use chainer.using_config')
+    argument.parse_kwargs(kwargs)
```
Comment (delta2323) [inline chainer/functions/connection/n_step_lstm.py]: This function call is for checking if there is no redundant arguments. As we cannot read the intension from function name, please add it to the comment.

File: chainer/functions/noise/dropout.py [hunk 3]
```diff
@@ -51,6 +60,11 @@ def dropout(x, ratio=.5):
     co-adaptation of feature detectors <https://arxiv.org/abs/1207.0580>`_.
 
     """
+    argument.check_unexpected_kwargs(
+        kwargs, train='train argument is not supported anymore. '
+        'Use chainer.using_config')
+    argument.parse_kwargs(kwargs)
```
Comment (delta2323) [inline chainer/functions/noise/dropout.py]: ditto

File: chainer/functions/noise/zoneout.py [hunk 4]
```diff
@@ -53,6 +62,11 @@ def zoneout(h, x, ratio=.5):
     Activations <https://arxiv.org/abs/1606.01305>`_.
 
     """
+    argument.check_unexpected_kwargs(
+        kwargs, train='train argument is not supported anymore. '
+        'Use chainer.using_config')
+    argument.parse_kwargs(kwargs)
```
Comment (delta2323) [inline chainer/functions/noise/zoneout.py]: ditto

File: chainer/links/model/vision/resnet.py [hunk 5]
```diff
@@ -164,6 +173,11 @@ def __call__(self, x, layers=['prob']):
 
         """
 
+        argument.check_unexpected_kwargs(
+            kwargs, test='test argument is not supported anymore. '
+            'Use chainer.using_config')
+        argument.parse_kwargs(kwargs)
```
Comment (delta2323) [inline chainer/links/model/vision/resnet.py]: ditto

File: chainer/links/connection/zoneoutlstm.py [hunk 6]
```diff
@@ -7,15 +7,20 @@
...
 
 
 class StatefulZoneoutLSTM(link.Chain):
 
-    def __init__(self, in_size, out_size, c_ratio=0.5, h_ratio=0.5):
+    def __init__(self, in_size, out_size, c_ratio=0.5, h_ratio=0.5, **kwargs):
+        argument.check_unexpected_kwargs(
+            kwargs, train='train argument is not supported anymore. '
+            'Use chainer.using_config')
+        argument.parse_kwargs(kwargs)
```
Comment (delta2323) [inline chainer/links/connection/zoneoutlstm.py]: ditto

File: chainer/links/connection/n_step_lstm.py [hunk 7]
```diff
@@ -82,6 +96,11 @@ def __call__(self, hx, cx, xs):
                 Each element ``xs[i]`` is a :class:`chainer.Variable` holding
                 a sequence.
         """
+        argument.check_unexpected_kwargs(
+            kwargs, train='train argument is not supported anymore. '
+            'Use chainer.using_config')
+        argument.parse_kwargs(kwargs)
```
Comment (delta2323) [inline chainer/links/connection/n_step_lstm.py]: ditto

File: chainer/links/connection/n_step_lstm.py [hunk 8]
```diff
@@ -48,7 +49,12 @@ class NStepLSTM(link.ChainList):
 
     """
 
-    def __init__(self, n_layers, in_size, out_size, dropout):
+    def __init__(self, n_layers, in_size, out_size, dropout, **kwargs):
+        argument.check_unexpected_kwargs(
+            kwargs, use_cudnn='use_cudnn argument is not supported anymore. '
+            'Use chainer.using_config')
+        argument.parse_kwargs(kwargs)
```
Comment (delta2323) [inline chainer/links/connection/n_step_lstm.py]: ditto

File: chainer/links/caffe/caffe_function.py [hunk 9]
```diff
@@ -179,6 +188,11 @@ def __call__(self, inputs, outputs, disable=()):
                 corresponding to elements of the  `outputs` argument.
 
         """
+        argument.check_unexpected_kwargs(
+            kwargs, train='train argument is not supported anymore. '
+            'Use chainer.using_config')
+        argument.parse_kwargs(kwargs)
```
Comment (delta2323) [inline chainer/links/caffe/caffe_function.py]: ditto

File: chainer/links/model/vision/resnet.py [hunk 10]
```diff
@@ -177,15 +191,25 @@ def __call__(self, x, layers=['prob']):
...
         transforms them to a proper variable. That is,
         it is also interpreted as a shortcut method that implicitly calls
         ``prepare`` and ``__call__`` functions.
 
+        .. warning::
+
+           ``test`` and ``volatile`` arguments are not supported anymore since
+           v2.
+           Instead, use ``chainer.using_config('train', train)`` and
+           ``chainer.using_config('enable_backprop', not volatile)``.
```
Comment (delta2323) [inline chainer/links/model/vision/resnet.py]: Add ", respectively"

File: chainer/links/model/vision/resnet.py [hunk 11]
```diff
@@ -201,6 +225,13 @@ def extract(self, images, layers=['pool5'], size=(224, 224)):
 
         """
 
+        argument.check_unexpected_kwargs(
+            kwargs, test='test argument is not supported anymore. '
+            'Use chainer.using_config',
+            volatile='volatile argument is not supported anymore. '
+            'Use chainer.using_config')
+        argument.parse_kwargs(kwargs)
```
Comment (delta2323) [inline chainer/links/model/vision/resnet.py]: ditto

File: chainer/links/model/vision/vgg.py [hunk 12]
```diff
@@ -168,6 +176,11 @@ def __call__(self, x, layers=['prob']):
 
         """
 
+        argument.check_unexpected_kwargs(
+            kwargs, test='test argument is not supported anymore. '
+            'Use chainer.using_config')
+        argument.parse_kwargs(kwargs)
```
Comment (delta2323) [inline chainer/links/model/vision/vgg.py]: ditto

File: chainer/links/model/vision/vgg.py [hunk 13]
```diff
@@ -181,15 +194,25 @@ def __call__(self, x, layers=['prob']):
...
         transforms them to a proper variable. That is,
         it is also interpreted as a shortcut method that implicitly calls
         ``prepare`` and ``__call__`` functions.
 
+        .. warning::
+
+           ``test`` and ``volatile`` arguments are not supported anymore since
+           v2.
+           Instead, use ``chainer.using_config('train', train)`` and
+           ``chainer.using_config('enable_backprop', not volatile)``.
```
Comment (delta2323) [inline chainer/links/model/vision/vgg.py]: Add ", respectively"

File: chainer/links/model/vision/vgg.py [hunk 14]
```diff
@@ -205,6 +228,13 @@ def extract(self, images, layers=['fc7'], size=(224, 224)):
 
         """
 
+        argument.check_unexpected_kwargs(
+            kwargs, test='test argument is not supported anymore. '
+            'Use chainer.using_config',
+            volatile='volatile argument is not supported anymore. '
+            'Use chainer.using_config')
+        argument.parse_kwargs(kwargs)
```
Comment (delta2323) [inline chainer/links/model/vision/vgg.py]: ditto

File: chainer/functions/connection/n_step_lstm.py (hunk 2 above)
Comment (unnonouno) [inline chainer/functions/connection/n_step_lstm.py]: OK. For this purpose, i'll make another method which only checks if kwargs is empty.

**Comment by delta2323** [comment]
Fix flake8 validation.

File: chainer/utils/argument.py [hunk 15]
```diff
@@ -11,3 +11,7 @@ def parse_kwargs(kwargs, *name_and_values):
         args = ', '.join(["'%s'" % arg for arg in kwargs.keys()])
         raise TypeError('got unexpected keyword argument(s) %s' % args)
     return tuple(values)
+
+
+def assert_kwargs_empty(kwargs):
+    return parse_kwargs(kwargs)
```
Comment (delta2323) [inline chainer/utils/argument.py]: I think we should not return nothing because we cannot infer what is returned from the function name.

File: chainer/functions/connection/n_step_lstm.py (hunk 2 above)
Comment (delta2323) [inline chainer/functions/connection/n_step_lstm.py]: Use `argument.assert_kwargs_empty(kwargs)` here, too.

File: chainer/functions/connection/n_step_lstm.py [hunk 16]
```diff
@@ -373,8 +374,10 @@ def _stack_weight(ws):
 
 
 def n_step_lstm(
-        n_layers, dropout_ratio, hx, cx, ws, bs, xs):
-    """Stacked Long Short-Term Memory function for sequence inputs.
+        n_layers, dropout_ratio, hx, cx, ws, bs, xs, **kwargs):
+    """n_step_lstm(n_layers, dropout_ratio, hx, cx, ws, bs, xs)
```
Comment (delta2323) [inline chainer/functions/connection/n_step_lstm.py]: Could you tell me why you write the API of this function in the first line of the docstring? Is that because `kwargs` should be empty?

Comment (unnonouno) [inline chainer/functions/connection/n_step_lstm.py]: `kwargs` is just for showing an error message when obsolete API is used. I think it is redundant to write in the document. How do you think?

Comment (delta2323) [inline chainer/functions/connection/n_step_lstm.py]: Sorry, I just did not know we can overwrite the API in the document when we write the API in the first line. I agree with we do not show `kwargs` in the document.

File: chainer/functions/noise/dropout.py [hunk 17]
```diff
@@ -33,13 +34,21 @@ def backward(self, x, gy):
         return gy[0] * self.mask,
 
 
-def dropout(x, ratio=.5):
-    """Drops elements of input variable randomly.
+def dropout(x, ratio=.5, **kwargs):
+    """dropout(x, ratio=.5):
```
Comment (delta2323) [inline chainer/functions/noise/dropout.py]: remove the colon

File: chainer/functions/noise/zoneout.py [hunk 18]
```diff
@@ -34,13 +35,21 @@ def backward(self, inputs, gy):
         return gy[0] * self.flag_h, gy[0] * self.flag_x,
 
 
-def zoneout(h, x, ratio=.5):
-    """Drops elements of input variable and sets to previous variable randomly.
+def zoneout(h, x, ratio=.5, **kwargs):
+    """zoneout(h, x, ratio=.5):
```
Comment (delta2323) [inline chainer/functions/noise/zoneout.py]: Remove the colon.

File: chainer/functions/normalization/batch_normalization.py [hunk 19]
```diff
@@ -318,7 +326,15 @@ def batch_normalization(x, gamma, beta, eps=2e-5, running_mean=None,
 
     .. seealso:: :class:`links.BatchNormalization`
 
-    """
+    """  # NOQA
```
Comment (delta2323) [inline chainer/functions/normalization/batch_normalization.py]: Why `# NOQA` is needed here?

Comment (unnonouno) [inline chainer/functions/normalization/batch_normalization.py]: Because the first line is too long and flake8 reports an error.

**Comment by delta2323** [comment]
LGTM
